coops-ui/src/components/Common/OfficeAddresses.vue

```diff
@@ -0,0 +1,813 @@
+            @click="updateAddress"
+          >
+            Update Addresses
+          </v-btn>
+          <v-btn id="reg-off-cancel-addr-btn" @click="cancelEditAddress">Cancel</v-btn>
+        </div>
+      </li>
+    </ul>
+  </v-card>
+</template>
+
+<script lang="ts">
+// Libraries
+import { Component, Emit, Prop, Watch, Mixins } from 'vue-property-decorator'
+import isEmpty from 'lodash.isempty'
+
+// Schemas
+import { addressSchema } from '@/schemas'
+
+// Components
+import BaseAddress from 'sbc-common-components/src/components/BaseAddress.vue'
+
+// Mixins
+import { AddressMixin, CommonMixin, EntityFilterMixin } from '@/mixins'
+
+// Interfaces
+import { BaseAddressObjIF, BcorpAddressIf, AddressIF } from '@/interfaces/address-interfaces'
+
+// Constants
+import { ADDRESSCHANGED } from '@/constants'
+
+// Enums
+import { EntityTypes } from '@/enums'
+
+@Component({
+  components: {
+    'delivery-address': BaseAddress,
+    'mailing-address': BaseAddress
+  }
+})
+export default class OfficeAddresses extends Mixins(AddressMixin, CommonMixin, EntityFilterMixin) {
+    /**
+     * Indicates whether the change button should be disabled or not
+     */
+    @Prop({ default: false })
+    readonly changeButtonDisabled: boolean
+
+    /**
+     * Addresses object from the parent page.
+     * If this is null then this is a new filing; otherwise these are the addresses from a draft filing.
+     * This will be emitted back to the parent page when the addresses are updated.
+     */
+    @Prop({ default: null })
+    readonly addresses: BcorpAddressIf | null
+
+    /**
+     * Registered Office address object passed in from the parent
+     */
+    @Prop({ default: null })
+    private registeredAddress: BaseAddressObjIF
+
+    /**
+     * Records Office address object passed in from the parent
+     */
+    @Prop({ default: null })
+    private recordsAddress: BaseAddressObjIF
+
+    // The two addresses that come from the store. These are used to reset the address.
+    private deliveryAddressOriginal = {} as AddressIF
+    private mailingAddressOriginal = {} as AddressIF
+
+    // The two addresses that come from the store. These are used to reset the address.
+    private recDeliveryAddressOriginal = {} as AddressIF
+    private recMailingAddressOriginal = {} as AddressIF
+
+    // The two addresses that are the current state of the BaseAddress components.
+    private deliveryAddress = {} as AddressIF
+    private mailingAddress = {} as AddressIF
+
+    // The two addresses that are the current state of the BaseAddress components.
+    private recDeliveryAddress = {} as AddressIF
+    private recMailingAddress = {} as AddressIF
+
+    // The two addresses for Registered Office where the above are stored prior to an edit. These allow a cancel to
+    // the address prior to edit, which if there was a prior edit will not be the data that originally
+    // came from the store.
+    private deliveryAddressTemp = {} as AddressIF
```

Comment:
Ah, so we have THREE separate addresses for reset purposes. Sigh...

Does the comment above make sense? I mean, if you cancel an edit, shouldn't we fall back to the address from the store? (I'm honestly not sure what the answer is.)

Reply:
Make perfect sense to remove these, i think the original thought process to prevent having to do another api call ( which was also redundant and was also removed ). Great comment!

Reply:
Thanks for the cleanup.